### PR TITLE
Remove deprecated flag --conntrack-max from kube-proxy

### LINKED ISSUE #78296: Remove deprecated kube-proxy flag --conntrack-max
The flag has been deprecated for many releases and can be removed.

The depreciation guide specifies that admin-only CLI flags can be removed after 6 months: https://kubernetes.io/docs/reference/using-api/deprecation-policy/

/kind cleanup
/sig network
/priority backlog

### PR DESCRIPTION
**Pull request by rikatz**
**What type of PR is this?** 
/kind cleanup

**What this PR does / why we need it**: This PR removes the deprecated --conntrack-max from kube-proxy and also the tests related to this flag

**Which issue(s) this PR fixes**:

Fixes #78296 

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
ACTION REQUIRED  The deprecated flag --conntrack-max has been removed from kube-proxy. Users of this flag should switch to --conntrack-min and --conntrack-max-per-core instead.
```

### CODE REVIEW

**Comment by vllry** [comment]
Cool, looks like you resolved the main issue! I'm curious how you figured it out. :)

**Comment by vllry** [comment]
@rikatz you can remove the first test case in `errorCases` in `TestValidateKubeProxyConntrackConfiguration`. The test fails because that configuration is now valid (it was specifically testing that the now-removed negative Max value would be rejected).

**Comment by rikatz** [comment]
> Cool, looks like you resolved the main issue! I'm curious how you figured it out. :)

YES! :D

Actually I've missed some Test Cases to be adjusted. They were generating the fuzzers wrongly :)

Also, before sending the PR I did a ``make clean_generated && make generated_files`` to regenerate all the ``zz`` files. After all, running a ``make verify`` gave to me no error (don't know how I missed this validation test, but it worked).

Now let's really cross fingers and hope all the CI passes successfully :)

**Comment by rikatz** [comment]
> @rikatz you can remove the first test case in `errorCases` in `TestValidateKubeProxyConntrackConfiguration`. The test fails because that configuration is now valid (it was specifically testing that the now-removed negative Max value would be rejected).

Missed this comment as I was deeply trying to figure out what went wrong ;( But tks for pointing that

**Comment by vllry** [comment]
This is flaking across the org, probably best to wait until tomorrow when the test-infra folks can fix it.

**Comment by rikatz** [comment]
Let me see if the merged fix of the test-infra works now

**Comment by rikatz** [comment]
No luck here trying to start the job again :/

**Comment by vllry** [comment]
Thanks! You can remove the "WIP".
/lgtm

/assign @thockin 
/assign @chuckha 
/assign @cmluciano

**Comment by thockin** [comment]
This needs an "action required" release note for users who are depending on this flag.

**Comment by rikatz** [comment]
> This needs an "action required" release note for users who are depending on this flag.

@thockin changed as asked. Is this fine or need some more info? Tks!!

**Comment by rikatz** [comment]
/uncc @thockin 
/uncc @cmluciano 
/uncc @chuckha 

@vllry I'm removing the reviewers again, as I'm solving the conflicts between a previous merge. After your new LGTM can we proceed?

Tks!!

**Comment by vllry** [comment]
@rikatz the merge looks bad - there was another PR that reorganized the flags, but in your merge diff they only shows deleted lines. Personally I prefer rebases over merges in k8s, as I've had a lot of trouble with merges and mysterious discrepancies.

**Comment by rikatz** [comment]
OK, trying again. I've removed the whole server.go file, and updated it with the the file in master + the updates of this PR

**Comment by vllry** [comment]
It may be easier to revert the merge, and do a rebase. The conflicts are fairly straightforward to solve.

**Comment by vllry** [comment]
Thanks @rikatz, looks good! The code freeze is imminent for 1.15, so this may have to wait until 1.16.

/assign @thockin
/assign @chuckha
/assign @cmluciano

File: pkg/proxy/apis/config/v1alpha1/defaults.go [hunk 1]
```diff
@@ -71,16 +71,15 @@ func SetDefaults_KubeProxyConfiguration(obj *kubeproxyconfigv1alpha1.KubeProxyCo
...
-	if obj.Conntrack.Max == nil {
-		// If ConntrackMax is *not* set, use per-core scaling.
-		if obj.Conntrack.MaxPerCore == nil {
-			obj.Conntrack.MaxPerCore = pointer.Int32Ptr(32 * 1024)
-		}
-		if obj.Conntrack.Min == nil {
-			obj.Conntrack.Min = pointer.Int32Ptr(128 * 1024)
-		}
+
+	// If ConntrackMax is *not* set, use per-core scaling.
```
Comment (dcbw) [inline pkg/proxy/apis/config/v1alpha1/defaults.go]: @vllry Is this comment still accurate?

Comment (vllry) [reply pkg/proxy/apis/config/v1alpha1/defaults.go]: I don't think it is.

Comment (thockin) [reply pkg/proxy/apis/config/v1alpha1/defaults.go]: +1

**Review by thockin — COMMENTED** [review]
The comment can be fixed in a followup, please.

/approve

**Comment by rikatz** [comment]
> The comment can be fixed in a followup, please.
> 
> /approve

OK, will wait until this gets merged and open another one following just to some comments cleanup.

Also when the issue gets automatically closed, will reopen it just to associate with the comment cleanup. Is that ok?

**Comment by rikatz** [comment]
@vllry As prow asked to rebase, I did it and also removed the unnecessary comment. Pinging you just to check if this is still good to enter into this milestone, otherwise no rush :)

**Comment by vllry** [comment]
/lgtm

But I believe we missed the boat for 1.15. Luckily there's no pressure to get this out, I just thought it would be good to do sooner rather than later.

**Comment by rikatz** [comment]
Got it. No problems at all. I’m leaving on vacations on Monday and will be away from computers until 18th, jun. Hope this doesn’t lock any other PRs, will try anyway to follow here and make the necessary changes ASAP. 

Thanks for the patience and the mentorship, this community is amazing!

**Pull request merged**